atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/QosMetrics.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.palantir.atlasdb.util.MetricsManager;
+
+public class QosMetrics {
+
+    private final MetricsManager metricsManager = new MetricsManager();
+
+    private final Meter readRequestCount;
+    private final Meter writeRequestCount;
+    private final Histogram bytesRead;
+    private final Histogram bytesWritten;
+
+    public QosMetrics() {
+        readRequestCount = metricsManager.registerMeter(QosMetrics.class, "numReadRequests");
+        writeRequestCount = metricsManager.registerMeter(QosMetrics.class, "numWriteRequests");
+
+        //TODO: The histogram should have a sliding window reservoir.
+        bytesRead = metricsManager.registerHistogram(QosMetrics.class, "bytesRead");
+        bytesWritten = metricsManager.registerHistogram(QosMetrics.class, "bytesWritten");
```

Comment:
if we go sliding time window reservoir, definitely need to consider what the memory bounds overhead of that will require.

Reply:
What will the sliding window reservoir give us? Does that affect the cumulative counts or just things like m15?

Reply:
I want to keep the metrics info for the last 30s or so around. The exponentially decaying reservoir  `.getSnapshot` gives us quantiles, which are not easy to reason about.

I was thinking if we could rate limit based on simple metrics like total bytes read/written in the last 30s? These are difficult to extract from an `exponentialDecayResorvoir` while `slidingWindowReservoir.getSnapshot` gives us this information for free.

Reply:
Do we really need the Snapshot even?  We definitely want the sum of X in a window but its not clear to me that we want the quantiles, which is the only reason we'd need to keep the Reservoir around.

Reply:
The question we're interested in here is: what is the current rate at which I'm reading / writing data? This can probably be answered by just having a cumulative count (`Counter` probably works, though `Meter` gives some extra info like mean rate and m5 rate for free).

It would also be nice to have a histogram showing average amount of data per query, but don't think we have a need for that now for QoS.

Reply:
Switched to meters here.